rir/src/compiler/opt/cleanup_framestate.cpp

```diff
@@ -7,25 +8,13 @@
 namespace rir {
 namespace pir {
 
-void CleanupFrameState::apply(RirCompiler&, Closure* function) const {
+void CleanupFramestate::apply(RirCompiler&, Closure* function) const {
     auto apply = [](Code* code) {
         Visitor::run(code->entry, [&](Instruction* i) {
             if (auto call = CallInstruction::CastCall(i)) {
                 call->clearFrameState();
             }
         });
-
```

Comment:
any instruction which has no effect and is unused is always removed by the normal cleanup phase. So there is no point in having a dedicated framestate cleanup phase. The only thing that needs to be done is remove framestates from calls, since there they are just metadata and not actual input to the instruction. Thus they generate sort of a "fake" dependency.

The reason we actually did explicitly remove fs in the past was that it was marked as effectful (which makes no sense for a metadata only instruction). after fixing that, fs gets removed automatically.